tstate/tstate_view.go

```diff
@@ -49,14 +52,31 @@ type TStateView struct {
 	writes      map[string]uint16
 }
 
-func (ts *TState) NewView(scope set.Set[string], storage map[string][]byte) *TStateView {
+func (ts *TState) NewView(scope set.Set[state.Key], storage map[string][]byte) *TStateView {
+	// TODO: set length to be the number of keys for each type
+	readScope := set.NewSet[string](len(scope))
+	writeScope := set.NewSet[string](len(scope))
+	rwScope := set.NewSet[string](len(scope))
+	for k := range scope {
+		switch {
+		case k.HasMode(state.Read):
+			readScope.Add(k.Name)
+		case k.HasMode(state.Write):
+			writeScope.Add(k.Name)
+		case k.HasMode(state.Read) && k.HasMode(state.Write):
+			rwScope.Add(k.Name)
+		}
+	}
+
 	return &TStateView{
 		ts:                 ts,
 		pendingChangedKeys: make(map[string]maybe.Maybe[[]byte], len(scope)),
 
 		ops: make([]*op, 0, defaultOps),
 
-		scope:        scope,
+		readScope:    readScope,
+		writeScope:   writeScope,
```

Comment:
I believe we should be checking for this scope in `Insert` call, I think the current code is still checking against the read + readwrite scope map when it should actually be checking against write + readwrite. We should add a test for this to make sure scopes are honored

Reply:
I added additional tests in `tstate_test.go` where I tested it on GetValue/Insert/Remove using different key permissions (i.e., No perms/Read/Write/RW), and the scopes look like they are honored (that is, trying to Insert a key that was specified to only be read or had no permissions would error, etc.)